### DISCUSSION TITLE
Balancing weights using (ne - oe).mean() vs base res?

### DISCUSSION THREAD

**Discussion by wkitlasten** [pypest/pyemu#408] [Ideas]
I notice the pst.adjust_weights() method ultimately uses pcf.base.rei to adjust the weights in the ies realm. In my case, those residuals are quite different than the mean residual (which I assume is appropriately calculated using (ne - oe).mean()?). Since the empirical jco is based on the ensemble mean, does it make sense to adjust the weights based on the ensemble based mean residual rather than the base realization? Or an I missing something (again)?

**Comment by jtwhite79** [comment]
There is a lot of subjectivity is all this weight stuff..and using the res file was easiest...you can always overwrite the values in pst.res with the ensemble mean values tho.  We're you thinking of something else?

**Comment by jtwhite79** [comment]
Also FWIW the latest version of ies does this kind of weight adjustment internally and uses the mean residuals...

**Reply by wkitlasten** [reply]
Ha, cool. I see the `ies_weight_ensemble()` but it seems like that is something different? Any hints?

**Comment by jtwhite79** [comment]
[https://github.com/usgs/pestpp/blob/develop/documentation/pestpp_users_manual.md#9210-internal-weight-adjustment](https://github.com/usgs/pestpp/blob/develop/documentation/pestpp_users_manual.md#9210-internal-weight-adjustment) 

It's in the version 5.2 beta release candidate. 

If you try it out, keen to hear how it goes - I've been using it quite a bit lately...

**Reply by wkitlasten** [reply]
Sounds cool. I took the "burdensome and potentially error prone" approach for now, but will give it a shot in a few days. The idea of having 'tags' in pestpp is cool.
